### PR TITLE
[GR-30453] Added an API for uninterruptible read and write operations on files.

### PR DESCRIPTION
**Pull request by christianhaeubl**

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixRawFileOperationSupport.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    private static int parseMode(FileAccessMode mode) {
+        if (mode == FileAccessMode.READ) {
+            return Fcntl.O_RDONLY();
+        } else if (mode == FileAccessMode.READ_WRITE) {
+            return Fcntl.O_RDWR() | Fcntl.O_CREAT();
+        } else if (mode == FileAccessMode.WRITE) {
+            return Fcntl.O_WRONLY() | Fcntl.O_CREAT();
+        }
+
+        throw new IllegalArgumentException("Illegal file access mode '" + Objects.toString(mode) + "'.");
```
Comment (rkennke) [inline substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixRawFileOperationSupport.java]: I don't think that you need Objects.toString() here, and can simply use 'mode'. Unless you want it formatted as octal number, but then you need more than Objects.toString() ;-)

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/os/RawFileOperationSupport.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    public interface RawFileDescriptor extends WordBase {
+    }
+
+    /**
+     * The file access modes that can be used when opening/creating a file.
+     */
+    public enum FileAccessMode {
+        READ,
+        READ_WRITE,
+        WRITE;
```
Comment (rkennke) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/os/RawFileOperationSupport.java]: Is that semicolon needed here?

**Review by rkennke — APPROVED** [review]
Looks good to me, only small remarks.

**Review by jiekang — APPROVED** [review]

**Review by rkennke — APPROVED** [review]

**Comment by christianhaeubl** [comment]
This was merged to graal master in the meanwhile (with a couple more minor cleanups). So, I am closing this PR.

**Pull request closed without merging**